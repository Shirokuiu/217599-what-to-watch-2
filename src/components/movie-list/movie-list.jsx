import React from "react";
import PropTypes from "prop-types";

import Movie from "../movie/movie";

export const MovieList = (props) => {
  const {films, onMovieClick, onMoviePageClick} = props;

  return <div className="catalog__movies-list">
    {films.map((film, idx) => (
      <Movie
        film={film}
        id={idx}
        onMovieClick={onMovieClick}
        onMoviePageClick={onMoviePageClick}
        key={film.id}
      />
    ))}
  </div>;
};

MovieList.propTypes = {
  films: PropTypes.array.isRequired,
  onMovieClick: PropTypes.func.isRequired,
  onMoviePageClick: PropTypes.func
};
