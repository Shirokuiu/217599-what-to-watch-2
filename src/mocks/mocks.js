export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
