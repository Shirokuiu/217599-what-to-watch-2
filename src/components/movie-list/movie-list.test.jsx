import React from "react";
import renderer from "react-test-renderer";

import {MovieList} from "./movie-list";

import {films} from "../../mocks/mocks";

it(`Is movie list rendered`, () => {
  const tree = renderer
    .create(<MovieList
      films={films}
      onMovieClick={jest.fn()}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
