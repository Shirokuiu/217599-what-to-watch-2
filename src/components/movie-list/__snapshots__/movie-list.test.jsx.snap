// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie list rendered 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-0-overview"
      >
        Fantastic Beasts
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-1-overview"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
</div>
`;
